logicrunner/handle_pending_finished.go

```diff
@@ -46,11 +46,10 @@ func (h *HandlePendingFinished) Present(ctx context.Context, _ flow.Flow) error
 	}
 
 	done, err := h.dep.WriteAccessor.Begin(ctx, flow.Pulse(ctx))
-	defer done()
-
 	if err != nil {
```

Comment:
Might be another error (not currently, but in future). It will be hard to debug.